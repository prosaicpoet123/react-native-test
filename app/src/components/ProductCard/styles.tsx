import { StyleSheet } from 'react-native'
import { colours, fontSizes } from '../../config/themes'

const styles = StyleSheet.create({
  productCard: {
    borderWidth: 1,
    borderColor: colours.grey4,
    padding: 4,
    minHeight: 300,
    backgroundColor: colours.white,
  },
  productCardLarge: {
    height: '100%',
  },
  productInfo: {
    padding: 4,
  },
  productInfoItem: {
    marginBottom: 8,
  },
  title: {
    fontWeight: 'bold',
  },
  titleLarge: {
    fontSize: fontSizes.large,
  },
  titleSmall: {
    fontSize: fontSizes.small,
  },
})

export default styles
