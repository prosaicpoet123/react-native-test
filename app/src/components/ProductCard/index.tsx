import React, { FC } from 'react'
import { Text, TouchableOpacity, View } from 'react-native'
import { IProduct } from '../../@Types/product'
import Availability from '../Availability'
import ProductImage from '../ProductImage'
import Ratings from '../Ratings'
import styles from './styles'

interface IProductCardProps {
  product: IProduct
  resolution?: number
  isDetails?: boolean
  onPress?: (x: IProduct) => void
}

type IProps = IProductCardProps

const ProductCard: FC<IProps> = ({
  product,
  resolution = 180,
  isDetails,
  onPress,
}) => {
  const {
    id,
    name,
    price,
    stock_status,
    rating,
    review_count,
    promotions,
    images,
  } = product

  return (
    <TouchableOpacity
      onPress={() => onPress && onPress(product)}
      disabled={isDetails}>
      <View style={[styles.productCard, isDetails && styles.productCardLarge]}>
        <ProductImage
          images={images}
          id={id}
          promotions={isDetails ? promotions : []}
          showThumbnails={isDetails}
          resolution={resolution}
        />
        <View style={styles.productInfo}>
          {!!name && (
            <Text
              style={[
                styles.productInfoItem,
                styles.title,
                isDetails ? styles.titleLarge : styles.titleSmall,
              ]}>
              {name}
            </Text>
          )}
          {!!isDetails && (
            <>
              {!!price && (
                <Text style={styles.productInfoItem}>{`Â£${price}`}</Text>
              )}
              {!!stock_status && (
                <View style={styles.productInfoItem}>
                  <Availability stock={stock_status} />
                </View>
              )}
              {!!rating && (
                <View style={styles.productInfoItem}>
                  <Ratings rating={rating} count={review_count} />
                </View>
              )}
            </>
          )}
        </View>
      </View>
    </TouchableOpacity>
  )
}

export default ProductCard
