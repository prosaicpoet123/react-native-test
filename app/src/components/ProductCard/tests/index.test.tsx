import React from 'react'
import renderer from 'react-test-renderer'
import ProductCard from '../'

describe('<ProductCard />', () => {
  it('renders correctly', () => {
    const product = {
      id: '000001',
      name: 'Example Product',
      price: 20,
      images: [],
    }
    const tree = renderer.create(<ProductCard product={product} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders correctly as details', () => {
    const product = {
      id: '000001',
      name: 'Example Product',
      price: 20,
      images: [],
    }
    const tree = renderer
      .create(<ProductCard product={product} isDetails />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
