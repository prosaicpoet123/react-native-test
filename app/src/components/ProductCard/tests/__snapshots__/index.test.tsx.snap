// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCard /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#CECECE",
          "borderWidth": 1,
          "minHeight": 300,
          "padding": 4,
        },
        undefined,
      ]
    }
  >
    <View>
      <View
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://images.hollandandbarrettimages.co.uk/productimages/HB/180/000001_A.jpg",
            }
          }
          style={
            Array [
              Object {
                "height": 180,
                "width": "100%",
              },
              Object {
                "height": 180,
              },
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "padding": 4,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "marginBottom": 8,
            },
            Object {
              "fontWeight": "bold",
            },
            Object {
              "fontSize": 14,
            },
          ]
        }
      >
        Example Product
      </Text>
    </View>
  </View>
</View>
`;

exports[`<ProductCard /> renders correctly as details 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#CECECE",
          "borderWidth": 1,
          "minHeight": 300,
          "padding": 4,
        },
        Object {
          "borderWidth": 0,
        },
      ]
    }
  >
    <View>
      <View
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://images.hollandandbarrettimages.co.uk/productimages/HB/180/000001_A.jpg",
            }
          }
          style={
            Array [
              Object {
                "height": 180,
                "width": "100%",
              },
              Object {
                "height": 180,
              },
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "flexWrap": "wrap",
            "marginVertical": 8,
          }
        }
      >
        <RCTScrollView
          data={Array []}
          disableVirtualization={false}
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={true}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          onContentSizeChange={[Function]}
          onEndReachedThreshold={2}
          onLayout={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          updateCellsBatchingPeriod={50}
          viewabilityConfigCallbackPairs={Array []}
          windowSize={21}
        >
          <View />
        </RCTScrollView>
      </View>
    </View>
    <View
      style={
        Object {
          "padding": 4,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "marginBottom": 8,
            },
            Object {
              "fontWeight": "bold",
            },
            Object {
              "fontSize": 22,
            },
          ]
        }
      >
        Example Product
      </Text>
      <Text
        style={
          Array [
            Object {
              "marginBottom": 8,
            },
            Object {
              "fontSize": 18,
              "fontWeight": "bold",
            },
          ]
        }
      >
        Â£20
      </Text>
    </View>
  </View>
</View>
`;
